### ISSUE TITLE
Projectiles: Null ref: Projectile.ProcessCollisionAndOrCollider

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#268]
Null ref in v0.0.15 with projectiles.

---

>Frame 26,266, EXCEPTION » NullReferenceException: Object reference not set to an instance of an object. >>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> Tallowmere2.Projectile.ProcessCollisionAndOrCollider (UnityEngine.Collision2D collision, UnityEngine.Collider2D collider, Tallowmere2.ProjectileCollisionPhase collisionPhase, System.Boolean useGameNonAllocRaycastHit, System.Boolean& hitParentCreatureShield) (at > <01557e0ef4c848a4891c26a1904d227f>:0) Tallowmere2.Projectile.Update () (at <01557e0ef4c848a4891c26a1904d227f>:0)
>
>Had just entered a new level; was using a bow and fighting a boomerager and a mage.
Also, had leaf bonus

---

>9:25:26, Frame 26,266, EXCEPTION » NullReferenceException: Object reference not set to an instance of an object. >>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> Tallowmere2.Projectile.ProcessCollisionAndOrCollider (UnityEngine.Collision2D collision, UnityEngine.Collider2D collider, Tallowmere2.ProjectileCollisionPhase collisionPhase, System.Boolean useGameNonAllocRaycastHit, System.Boolean& hitParentCreatureShield) (at <01557e0ef4c848a4891c26a1904d227f>:0) Tallowmere2.Projectile.Update () (at <01557e0ef4c848a4891c26a1904d227f>:0)
>
>Same game, much further through the level this time; lots of enemies, lightning ball equipped. Still had leaf bonus and now static bonus.

**Comment by ChrisNZL** [comment]
I have not been able to reproduce this with v0.0.15a.

In any case, have added a couple of null checks at the start of the Projectile.ProcessCollisionAndOrCollider method for v0.0.15b.

**Comment by ChrisNZL** [comment]
Still happening in v0.0.15b.

```
8:09:10, Frame 91431, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object.
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.Projectile.ProcessCollisionAndOrCollider (UnityEngine.Collision2D collision, UnityEngine.Collider2D collider, T2.ProjectileCollisionPhase collisionPhase, System.Boolean useGameNonAllocRaycastHit, System.Boolean& hitParentCreatureShield)
T2.Projectile.Update ()
```
Possible flow to reproduce:
1. Fight an enemy Outlander
2. Slay Outlander while their boomerang is flying towards player
3. Block the boomerang with shield

Additionally, might need to break down the `ProcessCollisionAndOrCollider` method into smaller methods to get a better stack trace.

**Comment by ChrisNZL** [comment]
Should be fixed in 0.0.15c.

Main cause seemed to be when boomerangs' owners would perish while boomerang was in mid-air. Might have been affecting other projectiles too. Added lots of further null checks in places.
